### ISSUE TITLE
Assert optimizers and loss functions in cmd args in train.py

### ISSUE THREAD

**Issue by vijayantajain** [vijayantajain/salicon_vgg16#18]

**Comment by vijayantajain** [comment]
Ugly but done
